go/collector.go

```diff
@@ -244,7 +329,7 @@ func (c *Collector) uploadToThunderstorm(info infoWithPath) (redo bool) {
 	if !extensionWanted && !magicHeaderWanted && (len(c.MagicHeaders) > 0 || len(c.FileExtensions) > 0) {
 		c.debugf("Skipping file %s with unwanted extension and magic header", info.path)
 		atomic.AddInt64(&c.Statistics.skippedFiles, 1)
```

Comment:
Similar to above - should this be part of the caller? If yes, what about the hash check below that skips files, but does not increase `skippedFiles`? Those files are handled weirdly right now since they don't increase any statistic.